central/config/datastore/datastore.go

```diff
@@ -42,6 +56,54 @@ var (
 
 type datastoreImpl struct {
 	store store.Store
+
+	publicConfigCache *expirable.LRU[string, *storage.PublicConfig]
+}
+
+// GetCachedPublicConfig returns the public part of the Central config.
+// The primary data source will be the cache and the secondary the database.
+func (d *datastoreImpl) GetCachedPublicConfig(ctx context.Context) (*storage.PublicConfig, error) {
+	var err error
+	publicConfig, found := d.publicConfigCache.Get(publicConfigKey)
+	if found && publicConfig != nil {
+		return publicConfig, nil
+	}
+
+	publicConfig, err = d.GetPublicConfig(ctx)
+	if err != nil {
+		return publicConfig, err
+	}
+	// The result of the cache addition is ignored as the information
+	// whether the cache did evict anything is irrelevant.
+	_ = d.publicConfigCache.Add(publicConfigKey, publicConfig)
+
+	return publicConfig, err
+}
+
+// GetPublicConfig returns the public part of the Central config
+func (d *datastoreImpl) GetPublicConfig(ctx context.Context) (*storage.PublicConfig, error) {
+	elevatedCtx := sac.WithGlobalAccessScopeChecker(
+		ctx,
```

Comment:
Do we need to use `ctx`? I would suggest ignoring it if we can, in that way the code is clearer that context is not relevant for access of public config.